src/main/java/nextstep/subway/domain/SectionLineUp.java

```diff
@@ -0,0 +1,171 @@
+
+    private static final int START_INDEX = 0;
+
+    @OneToMany(mappedBy = "line", orphanRemoval = true, cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<Section> sectionList = new ArrayList<>();
+
+    protected SectionLineUp() {
+    }
+
+    public void addSection(Section section) {
+        if (sectionList.isEmpty()) {
+            sectionList.add(section);
+            return;
+        }
+        validUnknownStation(section);
+        validSameSection(section);
+        add(section);
+    }
+
+    private void add(Section section) {
+        if (!isEndUpStation(section) && !isEndDownStation(section)) {
+            createInternal(section);
+            return;
+        }
+        sectionList.add(section);
+    }
+
+    public List<Section> getSectionsInOrder() {
+        Section firstSection = findFirstSection(sectionList.get(START_INDEX));
+        return searchSectionsByOrder(new ArrayList<>(), firstSection);
+    }
+
+    public Stations getStationsInOrder() {
+
+        Section firstSection = findFirstSection(sectionList.get(START_INDEX));
+        final List<Station> stations = searchStationsByOrder(new ArrayList<>(), firstSection);
+        return new Stations(stations.stream().distinct().collect(Collectors.toList()));
+    }
+
+    private boolean isEndUpStation(Section section) {
+        //  출발지가 같은 노선이 없고, 도착지에서 출발하는 노선은 있지만, 도착지로 향하는 노선이 없는 경우
+        return notHasSameUpStation(section) && notHasSameDownStation(section) && hasSameUpStationByDownStation(section);
+    }
+
+    private boolean isEndDownStation(Section section) {
+        //  출발지로 향하는 노선이 있지만, 출발지에서 출발하는 노선이 없는 경우
+        return hasSameDownStationByUpStation(section) && notHasSameUpStation(section);
+    }
+
+    private void createInternal(Section section) {
+        if (hasSameUpStation(section)) {
+            createInternalUpStation(section);
+            return;
+        }
+        if (hasSameDownStation(section)) {
+            createInternalDownStation(section);
+        }
+    }
+
+    private void createInternalUpStation(Section section) {
+        sectionList.stream().filter(streamSection -> streamSection.hasSameUpStation(section))
+                .findFirst()
+                .ifPresent(existSection -> {
+                    existSection.updateDistance(existSection.minusDistance(section));
+                    existSection.updateUpStation(section.getDownStation());
+                    sectionList.add(section);
+                });
+    }
+
+    private void createInternalDownStation(Section section) {
+        sectionList.stream().filter(streamSection -> streamSection.hasSameDownStation(section))
+                .findFirst()
+                .ifPresent(existSection -> {
+                    existSection.updateDistance(existSection.minusDistance(section));
+                    existSection.updateDownStation(section.getUpStation());
+                    sectionList.add(section);
+                });
+    }
```

Comment:
```suggestion
    private void createInternal(Section section) {
        findInternalMatchSection(section)
            .ifPresent(matchSection -> {
                matchSection.updateBy(section);
                sectionList.add(section);
            });
    }
```
이렇게 작성해보는건 어때요? 😀
`section`의 `distance`와 `upStation` 또는 `downStation`를 변경하는건 `section`에서 하도록 위임해줄수 있을것 같아요!

Reply:
해당 메서드가 하고자 하는 행위가 결국 `section` 내부 속성에 영향을 주는 것이기에,
피드백 주신 내용대로 `section`에 처리를 위임하는 것이 좋아보여요 !
피드백 주신 내용 반영하였습니다 
좋은 의견 감사합니다 ! 😀